File: dynamic_listings/lua/default/developing_mechanics/match3_newtiles.lua
local function createNewTiles()
    -- Our falling tiles list, will be used for tweening
    local falling_tiles = {}
    -- We scroll each column of the board
    for column_idx = 1, #matrix do
        column = matrix[column_idx]
        for row_idx = 1, #column do
            tile = column[row_idx]
            if (tile == nil) then
                local new_tile = Tile:random()
                new_tile.y = - 64 -- A value that is out of the board
                matrix[column][tile] = new_tile

                -- Add this tile to the falling tiles mapping
                falling_tiles[new_tile] = tile.y
            end
        end
    end
    return falling_tiles
end

File: dynamic_listings/python/default/developing_mechanics/match3_newtiles.py
def create_new_tiles():
    # Our falling tiles list, will be used for tweening
    falling_tiles: dict[Tile, int] = {}
    # We scroll each column of the board
    for column in matrix:
        for tile in column:
            if tile is None:
                new_tile: Tile = Tile.random()
                new_tile.y = - 64   # A value that is out of the board
                matrix[column][tile] = new_tile

                # Add this tile to the falling tiles mapping
                falling_tiles[new_tile] = tile.y
    return falling_tiles
